core/src/main/java/be/isach/ultracosmetics/util/ColorUtils.java

```diff
@@ -0,0 +1,65 @@
+package be.isach.ultracosmetics.util;
+
+import java.awt.Color;
+
+public class ColorUtils {
+    /**
+     * Generates a color in the rainbow sequence based on the current time. Use this method to get the smoothest
+     * transition between all colors of the rainbow.
+     *
+     * @return a Color representing the color for the current time.
+     */
+    public static Color getRainbowColor() {
+        return getRainbowColor(2);
+    }
+
+    /**
+     * Generates a color in the rainbow sequence based on the current time.
+     *
+     * @param secondsPerPhase number of seconds that each rainbow phase will last. There are a total of 6 phases. So,
+     *                        this number multiplied by 6 will be the total amount of time until the colors start
+     *                        repeating themselves.
+     *
+     * @return a Color representing the color for the current time.
+     */
+    public static Color getRainbowColor(double secondsPerPhase) {
+        long currentEpochMilliseconds = System.currentTimeMillis();
+        long currentEpochSeconds = currentEpochMilliseconds / 1000L;
+
+        double phase = currentEpochSeconds % (6 * secondsPerPhase);
+        phase = phase / secondsPerPhase;
+
+        double millisecondsToAdjustBy = currentEpochMilliseconds % (secondsPerPhase * 1000);
+
+        // There's a change of a specific number of rgb values in a single tick. This calculation is basically the
+        // current tick (or milliseconds) multiplied by 255 to get the rgb value.
+        int currentRotatingRgbValue = (int) (millisecondsToAdjustBy * 0.255 / secondsPerPhase);
+
+        int red = 0;
+        int green = 0;
+        int blue = 0;
+
+        // There are six phases in a rainbow when adjusting rgb values. This if-else encapsulates those six phases.
+        if (phase >= 0 && phase < 1) { // Red is at 255, Green goes to 255
```

Comment:
```suggestion
        if (phase == 0) { // Red is at 255, Green goes to 255
```
Why not just use equals here (and the related cases?) It seems to be logically equivalent unless I'm missing something.

Reply:
Yeah this is kinda odd. These need to be ranges now because the phase will be either equal or in between these two numbers. I wasn't sure how to make the phase still be an integer without the RGB value skipping around non-uniformally.

Reply:
Ah I see, then what about making phase an `int` so it rounds down? Like:
```diff
- double phase = currentEpochSeconds % (6 * secondsPerPhase);
- phase = phase / secondsPerPhase;
+ int phase = (currentEpochSeconds % (6 * secondsPerPhase)) / secondsPerPhase;
```

Reply:
Weird! I thought I had tried something like that and it hadn't worked. Guess I didn't know what I was doing! Haha. Looks like that suggestion works and then casting it to an `int` before it gets assigned:
```
int phase = (int) ((currentEpochSeconds % (6 * secondsPerPhase)) / secondsPerPhase)
```